lib/ui/workoutPlayer/workout_player.dart

```diff
@@ -0,0 +1,234 @@
+        color: Colors.black,
+        alignment: Alignment.center,
+        width: MediaQuery.of(context).size.width,
+        height: MediaQuery.of(context).size.height * 0.75,
+        child: AspectRatio(
+          aspectRatio: 16/9,
+          child: VideoPlayer(videoSrcUrl, UniqueKey(), handleVideoCompleted)
+        ),
+      ),
+      countDownTimer(context)
+    ]
+  );
+}
+
+Widget countDownTimer(BuildContext context) {
+  return Positioned(
+    top: 0,
+    left: MediaQuery.of(context).size.width / 30,
+    child: CircularCountDownTimer(
+      duration: 30,
+      initialDuration: 0,
+      width: MediaQuery.of(context).size.width / 30,
+      height: 200,
+      ringColor: Colors.grey,
+      fillColor: Colors.white,
+      strokeWidth: 3.0,
+      strokeCap: StrokeCap.round,
+      textStyle: const TextStyle(
+        fontSize: 20.0,
+        color: Colors.white,
+        fontWeight: FontWeight.bold
+      ),
+      textFormat: CountdownTextFormat.S,
+      isReverse: true,
+      isTimerTextShown: true,
+      autoStart: true,
+    )
+  );
+}
+
+Widget videoDescription() {
+  return const Text('This is the workout description', textAlign: TextAlign.left,);
+}
+
+Widget workoutTitle() {
+  return Container(
+    alignment: Alignment.centerLeft,
+    child: const Text(
+      'Title of Workout',
+      textAlign: TextAlign.left,
+      style: TextStyle(
+        fontWeight: FontWeight.bold,
+        color: Colors.black,
+        fontSize: 30
+      )
+    )
+  );
+}
+
+Widget exerciseList(BuildContext context, List<Exercise> exercises, Function setCurExercise, ItemScrollController _itemScrollController, curExerciseIndex) {
+  return exercises.isNotEmpty
+    ? ScrollablePositionedList.builder(
+        itemScrollController: _itemScrollController,
+        itemCount: exercises.length,
+        itemBuilder: (BuildContext context, int index) {
+          var itemKey = GlobalKey();
+          return GestureDetector(
+            key: itemKey,
+            child: exercise(context, exercises[index], curExerciseIndex == index),
+            onTap: () => {
+              setCurExercise(index),
+              playExercise(itemKey, index, _itemScrollController),
+            }
+          );
+        }
+      )
+      : Row(
+        children: const [
+          CircularProgressIndicator(),
+          Text("Loading")
+        ]
+      );
+}
+
+Future playExercise(itemKey, index, _scrollController) async {
+  await _scrollController.scrollTo(
+    index: index,
+    duration: Duration(milliseconds: 500),
+    curve: Curves.easeInOutCubic
+  );
+}
+
+Widget exercise(BuildContext context, Exercise exercise, bool isCurIndex) {
+  Duration duration = Duration(seconds: exercise.length);
+  String sDuration = "${duration.inMinutes.toString().padLeft(2, "0")}:${duration.inSeconds.remainder(60).toString().padLeft(2, "0")}";
+  return Container(
+    color: isCurIndex ? Colors.white : Colors.transparent,
+    child: ExpandablePanel(
+      header: Row(
```

Comment:
I think this is overflowing. Could you open a ticket and link it here to address this in the future?
![Capture](https://user-images.githubusercontent.com/31574865/144768166-e2f2f5fb-98a8-47db-9a0b-5c56f44b0f77.PNG)

Reply:
Good catch! Just opened a ticket: https://ubclaunchpad.atlassian.net/browse/WP-85?atlOrigin=eyJpIjoiZmU3MjM3YTBjZWY4NDU2ZGEwYzVjYmUwYTE4ZGRlZTIiLCJwIjoiaiJ9

Reply:
I am not having any overflowing issues on my side, both when im resizing the window and running it fullscreen